include/tests/propagator.hpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2021 CSCS, ETH Zurich
+ *               2021 University of Basel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*! @file
+ * @brief A Propagator class to manage the loop for each the timestep decoupled of the tests
+ *
+ * @author Sebastian Keller <sebastian.f.keller@gmail.com>
+ * @author Jose A. Escartin <ja.escartin@gmail.com>
+ */
+
+#pragma once
+
+#include <iostream>
+
+#include "../Timer.hpp"
+#include "../Task.hpp"
+#include "../ParticlesData.hpp"
+
+#include "cstone/domain/domain.hpp"
+
+#include "sph/timestep.hpp"
+
+using namespace cstone;
+using namespace sphexa;
+using namespace sphexa::sph;
+
+template <class I, class T, class Accelerator = CpuTag>
```

Comment:
These template arguments go to `hydroStep`, but as
`template<class DomainType, class ParticleDataType>`